File: w1_c/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_grid(int size);
int main(void)
{
    int n = get_height();
    print_grid(n);
}

int get_height()
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n > 8 || n < 1);
    return n;
}

void print_grid(int size)
{
    int i, j; // each j in different scope
    for (i = 0; i < size; i++)
    { // execute print space
        for (j = size - 1; j > i; j--)
        {
            printf(" ");
        }
        // execute print brick
        for (j = 0; j <= i; j++)
        {
            printf("#");
        }
        // excute print 2 spaces
        printf("  ");
        // execute print brick
        for (j = 0; j <= i; j++)
        {
            printf("#");
        }
        // execute a new line
        printf("\n");
    }
}

File: w6_python/sentimental-mario-more/mario.py
from cs50 import get_int

while True:
    n = get_int("Height: ")
    if n > 0 and n < 9:
        break

for i in range(n):
    # print() # (possible space) + brick + space2 + brick
    # print(" " * (n-1-i) + "#" * (i+1) + "  " + "#" * (i+1))
    space = " " * (n - 1 - i)
    brick = "#" * (i + 1)
    print(f"{space}{brick}  {brick}")
